crates/zkvm/lib/src/ecdsa/affine.rs

```diff
@@ -0,0 +1,219 @@
+use super::{
+    ECDSACurve, ECDSAPoint, Field, FieldElement, SP1AffinePointTrait, FIELD_BYTES_SIZE_USIZE,
+};
+
+use elliptic_curve::{
+    ff::Field as _,
+    group::GroupEncoding,
+    point::{AffineCoordinates, DecompactPoint, DecompressPoint},
+    sec1::{self, CompressedPoint, EncodedPoint, FromEncodedPoint, ToEncodedPoint},
+    subtle::{Choice, ConditionallySelectable, ConstantTimeEq, CtOption},
+    zeroize::DefaultIsZeroes,
+    FieldBytes, PrimeField,
+};
+use std::ops::Neg;
+
+#[derive(Clone, Copy, Debug)]
+pub struct AffinePoint<C: ECDSACurve> {
+    pub inner: C::SP1AffinePoint,
+}
+
+impl<C: ECDSACurve> AffinePoint<C> {
+    pub fn from_field_elements_unchecked(x: FieldElement<C>, y: FieldElement<C>) -> Self {
+        let mut x_slice = x.to_bytes();
+        let x_slice = x_slice.as_mut_slice();
+        x_slice.reverse();
+
+        let mut y_slice = y.to_bytes();
+        let y_slice = y_slice.as_mut_slice();
+        y_slice.reverse();
+
+        AffinePoint { inner: <C::SP1AffinePoint as ECDSAPoint>::from(x_slice, y_slice) }
+    }
+
+    /// # Panics
+    /// - if the point is the identity point.
```

Comment:
Two comments
- this doesn't panic when the identity point is input
- this doesn't work for curves where (0, 0) is a valid curve point (not the case for secp256k1 and secp256r1)